addons/account/models/account_invoice.py

```diff
@@ -1269,8 +1269,8 @@ def action_cancel(self):
         for inv in self:
             if inv.move_id:
                 moves += inv.move_id
-            if inv.payment_move_line_ids:
-                raise UserError(_('You cannot cancel an invoice which is partially paid. You need to unreconcile related payment entries first.'))
+            #unreconcile all journal items of the invoice, since the cancellation will unlink them anyway
+            inv.move_id.line_ids.filtered(lambda x: x.account_id.reconcile).remove_move_reconcile()
```

Comment:
@qdp-odoo Wouldn't it be ok to leave the warning just to alert the user in case there are payments in the invoice?
Otherwise proceed with Unreconciling of the Accruals.
Just to keep the same functionality.
Regards.

Reply:
well, I though that warning was not really welcome anyway... since you deliberately pressed the 'cancel' button and all the information (payments already done and outstanding credits when re-validated) are already visible on the invoice.

What do you think?

Reply:
@qdp-odoo from that PoV it looks awesome, It's less cumbersome, 
Go ahead then. 
👍 

Regards